Sources/Networking/HTTPClient/SimpleNetworkService.swift

```diff
@@ -0,0 +1,35 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  SimpleNetworkService.swift
+//
+//  Created by Jacob Zivan Rakidzich on 8/12/25.
+
+import Foundation
+
+/// A protocol representing a simple network service
+public protocol SimpleNetworkService {
+    /// Fetch data from the network
+    /// - Parameter url: The URL to fetch data from
+    /// - Returns: Data upon success
+    func data(from url: URL) async throws -> Data
+}
+
+extension URLSession: SimpleNetworkService {
+    /// Fetch data from the network
+    /// - Parameter url: The URL to fetch data from
+    /// - Returns: Data upon success
+    public func data(from url: URL) async throws -> Data {
+        let (data, response) = try await URLSession.shared.data(from: url)
```

Comment:
I am confused with this one. This is already an instance method, but it still uses `URLSession.shared`. Shouldn't it use self?

Reply:
Copy paste during refactor issues… thanks for catching that